src/mon/HealthMonitor.cc

```diff
@@ -885,3 +891,31 @@ void HealthMonitor::check_if_msgr2_enabled(health_check_map_t *checks)
     }
   }
 }
+
+void HealthMonitor::check_mon_crush_loc_stretch_mode(health_check_map_t *checks)
+{
+  // Check if the CRUSH location exists for all MONs
+  ceph_assert(mon.monmap->stretch_mode_enabled);
```

Comment:
Rather than asserting, why not just return (healthy) if stretch mode is off?
I put in a lot of asserts that caused crashes in my first stretch implementation, and assert crashes are better than data corruption, but not be in able to fail is even better!

Reply:
The reason I have the assert is because the callee is under the condition:
` if (mon.monmap->stretch_mode_enabled)`
so if we enter `check_mon_crush_loc_stretch_mode` and stretch mode is off, then something must have gone wrong. But I understand your point, we can potentially modify this such that we call check_mon_crush_loc_stretch_mode but then exit if stretch mode is not enabled, I can see this being a cleaner approach.